pennylane/transforms/adjoint_metric_tensor.py

```diff
@@ -157,78 +136,92 @@ def circuit(weights):
     shot simulations.
     """
     if isinstance(circuit, qml.tape.QuantumScript):
-        return _adjoint_metric_tensor_tape(circuit, device)
+        return _adjoint_metric_tensor_tape(circuit)
     if isinstance(circuit, (qml.QNode, qml.ExpvalCost)):
         return _adjoint_metric_tensor_qnode(circuit, device, hybrid)
 
     raise qml.QuantumFunctionError("The passed object is not a QuantumTape or QNode.")
 
 
-def _adjoint_metric_tensor_tape(tape, device):
+def _adjoint_metric_tensor_tape(tape):
     """Computes the metric tensor of a tape using the adjoint method and a given device."""
     # pylint: disable=protected-access
-    if device.shots is not None:
+    if tape.shots:
         raise ValueError(
             "The adjoint method for the metric tensor is only implemented for shots=None"
         )
+
+    if set(tape.wires) != set(range(tape.num_wires)):
+        wire_map = {w: i for i, w in enumerate(tape.wires)}
+        tape = qml.map_wires(tape, wire_map)
```

Comment:
Is this extending functionality or is it just necessary for compatibility with the new device methods? :)

Reply:
The functions in `devices.qubit` assume that the wire labels have been mapped to statevector dimensions.